QUARKUS-1676.md

```diff
@@ -0,0 +1,41 @@
+# QUARKUS-1676 - Add support for mTLS with OIDC Server
+
+JIRA link: https://issues.redhat.com/browse/QUARKUS-1676
+
+PR: https://github.com/quarkusio/quarkus/pull/23145
+
+Solving issue: https://github.com/quarkusio/quarkus/issues/19634#issuecomment-999435505
+
+Feature enables an OIDC client to present its TLS certificate when required by an OIDC server and thus have both parties
+of a TLS connection mutually authenticated.
+
+If the server requires client authentication then the client must present its own certificate to the server when
+connecting. The certificate is usually taken from keystore, thus mTLS is enabled when an OIDC client can be configured 
+to use client certificate from specified file and allows access to the file (by setting keystore pwd, possibly alias and
+alias pwd when there is more than key in a keystore). The OIDC Client now supports multiple formats (PKCS#12, JKS, ...)
+for a keystore file, therefore a keystore file type must be newly configurable (same goes for truststore).
+OIDC server choice is irrelevant in spite of the RFE as mutual handshake is standardized by [RFC 8846](https://datatracker.ietf.org/doc/html/rfc8446#appendix-E.1).
+
+## Scope of the testing
+
+Tests should verify:
+- User is able to add certificates to keystore and truststore to app run in a Java mode and a Native mode
+- Certificates and all relevant configurable options are picked up and used correctly by OIDC client
+- OIDC client and server communication is secured by mTLS when OIDC server requires so
+- Keycloak and RH SSO (latest and stable version).
+
+### Impact on testsuites and testing automation:
+- Ensure this coverage works fine on DEV mode
+- Ensure this coverage works fine on Unix and Windows environments (for coverage that does not need Docker as this is a limitation in Windows)
+
+## Getting familiar with the feature
+Following actions were taken to ensure familiarity:
+- Ensure documentation provides clear explanation on configuration options
+- Ensure documentation uses standard terminology and config options are relatable to Ver.x core options used underneath
+- Understand limitations of implementation
+- Check keystore types (namely JKS, PKCS12; other proprietary keystore types should be skipped)
+- Ensure good user experience and simplicity of use
+
+## Advanced topics for test development
+- Dev Services For Keycloak currently (Quarkus 2.8.0. Final) don't support mTLS (as requiring client
```

Comment:
This should maybe have its own issue linked to RFE in Jira, for developer experience

Reply:
I described better issue with "Dev Services for Keycloak", however I haven't filled an RFE yet. I am honestly not sure if the issue is big enough, it looks to me like whoever wants to use mTLS with Dev Services for Keycloak can simply build his own image (especially as you always want to run your development against Keycloak version used in production, so you have to set image version manually anyway).

However I am happy to create a ticket if you think it's convenient!

Michal